Adding documentation for: Zero shot object detection task (#435)
huggingface/huggingface.js: packages/tasks/src/tasks/zero-shot-object-detection/about.md

> ## About the task

Reviewer: ```suggestion
```
This header is added by default, no need to add

> Zero-shot object detection is a computer vision task to detect objects and its class in images, without any prior training or knowledge of the classes.

Reviewer: You can add this part to `summary` section in `data.ts`

> Zero shot object detection works by transferring knowledge learnt during training of one model, to classify novel classes that was not present in the training data.

Reviewer: We can also append it to above text and directly write to `summary` instead

Author: Should I remove both parts in the `about.md` file and instead write them in the `summary` of `data.ts` or remove keep them and add them to the `data.ts` file?

## Use Cases

### Object Search

> Zero-shot Object Detection models are used in image search. Smartphones use Object Detection models to detect entities (such as specific places or objects) and allow the user to search for the entity on the Internet.

Reviewer: ```suggestion
Zero-shot object detection models can be used in image search. Smartphones use zero-shot object detection models to detect entities (such as specific places or objects) and allow the user to search for the entity on the internet.
```

### Object Counting

> Zero-shot Object Detection models are used to count instances of objects in a given image, this can include counting the objects in warehouses or stores, or counting the number of visitors in a store. They are also used to manage crowds at events to prevent disasters.

Reviewer: ```suggestion
Zero-shot object detection models are used to count instances of objects in a given image. This can include counting the objects in warehouses or stores or the number of visitors in a store. They are also used to manage crowds at events to prevent disasters.
```

## Inference

> You can infer with Zero-shot Object Detection models through the `zero-shot-object-detection` pipeline. When calling the pipeline you just need to specify a path or http link to an image and the candidate labels.

Reviewer: ```suggestion
You can infer with zero-shot object detection models through the `zero-shot-object-detection` pipeline. When calling the pipeline, you just need to specify a path or HTTP link to an image and the candidate labels.
```

```python
detector = pipeline(model="google/owlvit-base-patch32", task="zero-shot-object-detection")

predictions = detector(
    image,
    candidate_labels=["a photo of a cat", "a photo of a dog"],
)

# [{'score': 0.95,
#   'label': 'a photo of a cat',
#   'box': {'xmin': 180, 'ymin': 71, 'xmax': 271, 'ymax': 178}},
#   ...
# ]
```

# Useful Resources

> - [Zero-shot Object detection task guide](https://huggingface.co/tasks/zero-shot-object-detection)

Reviewer: ```suggestion
- [Zero-shot object detection task guide](https://huggingface.co/tasks/zero-shot-object-detection)
```

## About the task

Zero-shot object detection is a computer vision task to detect objects and its class in images, without any prior training or knowledge of the classes.

Zero shot object detection works by transferring knowledge learnt during training of one model, to classify novel classes that was not present in the training data.

## Use Cases

### Object Search

> Zero-shot object detection models can be used in image search. Smartphones use zero-shot object detection models to detect entities (such as specific places or objects) and allow the user to search for the entity on the internet.

Reviewer 2: ```suggestion
Zero-shot object detection models can be used in image search. Smartphones, for example, use zero-shot object detection models to detect entities (such as specific places or objects) and allow the user to search for the entity on the internet.
```

### Object Counting

Zero-shot object detection models are used to count instances of objects in a given image. This can include counting the objects in warehouses or stores or the number of visitors in a store. They are also used to manage crowds at events to prevent disasters.

## Inference

You can infer with zero-shot object detection models through the `zero-shot-object-detection` pipeline. When calling the pipeline, you just need to specify a path or HTTP link to an image and the candidate labels.

```python
detector = pipeline(model="google/owlvit-base-patch32", task="zero-shot-object-detection")

predictions = detector(

>     image,

Reviewer 2: Would it be possible to initialize the image before use?

Author: I added the import for the `pipeline` and the image initialization using `Pillow`.

    candidate_labels=["a photo of a cat", "a photo of a dog"],
)

# [{'score': 0.95,
#   'label': 'a photo of a cat',
#   'box': {'xmin': 180, 'ymin': 71, 'xmax': 271, 'ymax': 178}},
#   ...
# ]
```

# Useful Resources

> - [Zero-shot object detection task guide](https://huggingface.co/tasks/zero-shot-object-detection)

Reviewer 2: ```suggestion
- [Zero-shot object detection task guide](https://huggingface.co/docs/transformers/tasks/zero_shot_object_detection)
```

Isn't the other URL the link to this same page?
